### PR TITLE
SUS-691 add migration script as a Task

### PR DESCRIPTION
**Pull request by sqreek**
This PR add job that will handle whole process of migration SMW to new storage.

@mixth-sense @macbre

### CODE REVIEW

File: extensions/wikia/SemanticMediaWiki/maintenance/SMW_MigrationJob.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+	public function run($cityId) {
+		$this->cityId = $cityId;
+		if (!$this->refreshPageProperties()) {
+			return false;
+		};
+		if (!$this->refreshPages()) {
+			return false;
+		}
+		return $this->setDb();
```
Comment (macbre) [inline extensions/wikia/SemanticMediaWiki/maintenance/SMW_MigrationJob.php]: `switchToNewStorage` instead?

File: extensions/wikia/SemanticMediaWiki/maintenance/SMW_refreshData.php [hunk 2]
```diff
@@ -69,6 +69,8 @@
 	$delay = false;
 }
 
+$taskId = isset($options['task_id'])?$options['task_id']:'';
+
```
Comment (macbre) [inline extensions/wikia/SemanticMediaWiki/maintenance/SMW_refreshData.php:73]: :+1:

File: extensions/wikia/SemanticMediaWiki/maintenance/SMW_MigrationJob.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+ *
+ * This job run SMW_refreshData that is migrate Storage for smw from store2 to store3 version
+ *
+ */
+
+use Wikia\Tasks\Tasks\BaseTask;
+
+class SMW_MigrationJob extends BaseTask {
+
+	var $cityId;
```
Comment (macbre) [inline extensions/wikia/SemanticMediaWiki/maintenance/SMW_MigrationJob.php]: Let's use a modern PHP here :)

``` php
private $cityId;
```

File: extensions/wikia/SemanticMediaWiki/maintenance/SMW_refreshData.php [hunk 4]
```diff
@@ -152,6 +176,20 @@
 		if ( $num_files % 100 === 0 ) { // every 100 pages only
 			$linkCache->clear(); // avoid memory leaks
 		}
+
+		# Wikia Change - begin
+		if ($num_files % 1000 === 0) {
+			Wikia\Logger\WikiaLogger::instance()->info('SMW_refreshData.php - process', [
+				'wikia_dbname' => $wgDBname,
+				'wikia_id' => $wgCityId,
```
Comment (macbre) [inline extensions/wikia/SemanticMediaWiki/maintenance/SMW_refreshData.php]: The two fields above are always logged as a part of a context:

```
@fields.wiki_dbname     kancolle
@fields.wiki_id     813333
```

File: extensions/wikia/SemanticMediaWiki/maintenance/SMW_runRefreshDataUsingMigrationJob.php [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+require_once ( getenv( 'MW_INSTALL_PATH' ) !== false
+	? getenv( 'MW_INSTALL_PATH' ) . "/maintenance/commandLine.inc"
+	: dirname( __FILE__ ) . '/../../../../maintenance/commandLine.inc' );
+
+global $wgCityId;
+
+$task = new SMW_MigrationJob();
+(new \Wikia\Tasks\AsyncTaskList())
+	->wikiId($wgCityId)
+	->add($task->call('run', $wgCityId))
```
Comment (macbre) [inline extensions/wikia/SemanticMediaWiki/maintenance/SMW_runRefreshDataUsingMigrationJob.php]: I'm wondering if we actually need to pass `city_id` in both place (as `wikiId` task param and the called method itself). Maybe just pick one?

**Comment by macbre** [comment]
:+1:

**Comment by ludwikkazmierczak** [comment]
Tested and Accepted

**Pull request merged**